lib/adapters/http/read.js

```diff
@@ -1,13 +1,94 @@
+'use strict';
 var _ = require('underscore');
 var contentType = require('content-type');
-var Juttle = require('../../runtime').Juttle;
+var AdapterRead = require('../../runtime/adapter-read');
 var parsers = require('../parsers');
+var errors = require('../../errors');
 var request = require('request');
+var Promise = require('bluebird');
+
+class Iterator {
+    constructor(adapter, req, res) {
+        this.points = [];
+        this.done = false;
+
+        var format;
+        try {
+            format = contentType.parse(res).type.split('/')[1];
```

Comment:
Know this was just a copy-paste but I'm confused by the logic...you take the content type of `res` and then use that to get a parser that you use on `req`? Seems like this should be `contentType.parse(req)`.